File: AtCoder/ACP/BootCamp4b/Medium100_2/python/B_Exactly_N_points.py
MAX_N = 10**7
N = int(input())
ub = MAX_N + 1
lb = 0
while ub - lb > 1:
    md = (lb + ub) // 2
    if md * (md + 1) // 2 >= N:
        ub = md
    else:
        lb = md
for i in range(1, ub + 1):
    if i == ub * (ub + 1) // 2 - N:
        continue
    print(i)

File: AtCoder/ACP/BootCamp4b/Medium100_2/cpp/B_Exactly_N_points.cpp
#include <bits/stdc++.h>
#define MAX_N 10'000'000
using namespace std;
typedef long long ll;

ll N;

void solve() {
  ll lb = 0;
  ll ub = MAX_N + 1;
  while (ub - lb > 1) {
    ll md = (ub + lb) / 2;
    if (md * (md + 1) / 2 >= N) {
      ub = md;
    } else {
      lb = md;
    }
  }
  for (int i = 1; i <= ub; ++i) {
    if (i == ub * (ub + 1) / 2 - N) {
      continue;
    }
    printf("%d\n", i);
  }
}

int main() {
  cin >> N;
  solve();
}